databricks_cli/dbfs/api.py

```diff
@@ -97,20 +115,80 @@ def file_exists(self, dbfs_path, headers=None):
             raise e
         return True
 
+    @retry(wait=wait_random_exponential(multiplier=EXPONENTIAL_BACKOFF_MULTIPLIER, 
+           max=MAX_SECONDS_WAIT), retry=retry_if_exception_type(RateLimitException), 
+           stop=stop_after_attempt(MAX_RETRY_ATTEMPTS))
+    def get_status_with_retries(self, dbfs_path, headers=None):
+        try:
+            return self.client.get_status(dbfs_path, headers=headers)
+        except HTTPError as e:
+            if e.response.status_code == 429:
+                click.echo("Rate Limit Exceeded. Trying again.")
+                raise RateLimitException
+            raise e
+
     def get_status(self, dbfs_path, headers=None):
-        json = self.client.get_status(dbfs_path.absolute_path, headers=headers)
+        json = self.get_status_with_retries(dbfs_path.absolute_path, headers=headers)
         return FileInfo.from_json(json)
 
+    @retry(wait=wait_random_exponential(multiplier=EXPONENTIAL_BACKOFF_MULTIPLIER, 
+           max=MAX_SECONDS_WAIT), retry=retry_if_exception_type(RateLimitException), 
+           stop=stop_after_attempt(MAX_RETRY_ATTEMPTS))
+    def create_with_retries(self, dbfs_path, overwrite, headers):
+        try:
+            return self.client.create(dbfs_path, overwrite, headers=headers)
+        except HTTPError as e:
+            if e.response.status_code == 429:
+                click.echo("Rate Limit Exceeded. Trying again.")
+                raise RateLimitException
+            raise e
+
+    @retry(wait=wait_random_exponential(multiplier=EXPONENTIAL_BACKOFF_MULTIPLIER, 
+           max=MAX_SECONDS_WAIT), retry=retry_if_exception_type(RateLimitException), 
+           stop=stop_after_attempt(MAX_RETRY_ATTEMPTS))
+    def add_block_with_retries(self, handle, contents, headers):
+        try:
+            self.client.add_block(handle, b64encode(contents).decode(), headers=headers)
+        except HTTPError as e:
+            if e.response.status_code == 429:
+                click.echo("Rate Limit Exceeded. Trying again.")
+                raise RateLimitException
+            raise e
+
+    @retry(wait=wait_random_exponential(multiplier=EXPONENTIAL_BACKOFF_MULTIPLIER, 
+           max=MAX_SECONDS_WAIT), retry=retry_if_exception_type(RateLimitException), 
+           stop=stop_after_attempt(MAX_RETRY_ATTEMPTS))
+    def close_with_retries(self, handle, headers):
+        try:
+            self.client.close(handle, headers=headers)
+        except HTTPError as e:
+            if e.response.status_code == 429:
+                click.echo("Rate Limit Exceeded. Trying again.")
+                raise RateLimitException
+            raise e
+
     def put_file(self, src_path, dbfs_path, overwrite, headers=None):
-        handle = self.client.create(dbfs_path.absolute_path, overwrite, headers=headers)['handle']
```

Comment:
Let's keep passing `dbfs_path.absolute_path` instead of changing it to `dbfs_path` to avoid moving extra logic in the `_with_retries` methods. See my comments below.

Reply:
Makes sense. Made them consistent.